File: backend/src/main/java/com/ll/nbe342team8/domain/cart/dto/CartResponseDto.java
package com.ll.nbe342team8.domain.cart.dto;

import com.ll.nbe342team8.domain.cart.entity.Cart;

public record CartResponseDto(
        Long bookId,
        int quantity,
        String title,
        int price,
        String coverImage
) {

    public static CartResponseDto from(Cart cart) {
        return new CartResponseDto(
                cart.getBook().getId(),
                cart.getQuantity(),
                cart.getBook().getTitle(),
                cart.getBook().getPricesSales(),
                cart.getBook().getCoverImage()
        );
    }
}

File: backend/src/main/java/com/ll/nbe342team8/domain/cart/dto/CartResponseDto.kt
package com.ll.nbe342team8.domain.cart.dto

import com.ll.nbe342team8.domain.cart.entity.Cart

data class CartResponseDto(
    val bookId: Long,
    val quantity: Int,
    val title: String,
    val price: Int,
    val coverImage: String
) {
    companion object {
        @JvmStatic
        fun from(cart: Cart): CartResponseDto {
            val book = cart.book
            val id = book.id ?: throw IllegalStateException("Book id is null")
            return CartResponseDto(
                bookId = id,
                quantity = cart.quantity,
                title = book.title,
                price = book.pricesSales,
                coverImage = book.coverImage
            )
        }
    }
}
